waku/v2/node/storage/message/waku_message_store.nim

```diff
@@ -51,6 +54,36 @@ proc messageCount(db: SqliteDatabase): MessageStoreResult[int64] =
     return err("failed to count number of messages in DB")
   ok(numMessages)
 
+
+proc getOldestDbReceiverTimestamp(db: SqliteDatabase): MessageStoreResult[int64] =
+  var oldestReceiverTimestamp: int64
+  proc handler(s: ptr sqlite3_stmt) =
+    oldestReceiverTimestamp = column_timestamp(s, 0)
+  let query = "SELECT receiverTimestamp FROM " & TABLE_TITLE & " ORDER BY receiverTimestamp LIMIT 1;";
```

Comment:
Is there a difference in performance between SELECT ... ORDER BY ... LIMIT 1 and: 
```suggestion
  let query = "SELECT MIN(receiverTimestamp) as minReceiverTimestamp FROM " & TABLE_TITLE;
```

Reply:
`MIN` is more efficient here. I  uses the primary key for searching. Thank you :). And its also more readable.

Ill do 

```nim
let query = "SELECT MIN(receiverTimestamp)  FROM " & TABLE_TITLE;
```

For now, I just wanted the retention timestamp feature to work. There is quite a lot in the code that could be improved / made more efficient  :sweat_smile: 
The goal is to have the feature in the next release, so that it can be tested to get earlier feedback.